source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -272,111 +272,183 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
-          // message.
-          if (absl::SimpleAtoi(headers_->getContentLengthValue(), &content_length) &&
-              content_length != common_response.body_mutation().body().size()) {
-            return absl::InternalError(
-                "mismatch between content length and the length of the mutated body");
-          }
-        }
-        ENVOY_LOG(debug, "Applying body response to buffered data. State = {}",
-                  static_cast<int>(callback_state_));
-        modifyBufferedData([&common_response](Buffer::Instance& data) {
-          MutationUtils::applyBodyMutations(common_response.body_mutation(), data);
-        });
-      }
-      clearWatermark();
-      onFinishProcessorCall(Grpc::Status::Ok);
-      should_continue = true;
-    } else if (callback_state_ == CallbackState::StreamedBodyCallback) {
-      absl::StatusOr<bool> result = handleBodyInStreamedState(common_response);
-      if (!result.ok()) {
-        return result.status();
-      }
-      should_continue = *result;
-    } else if (callback_state_ == CallbackState::BufferedPartialBodyCallback) {
-      // Apply changes to the buffer that we sent to the server
-      Buffer::OwnedImpl chunk_data;
-      auto chunk = dequeueStreamingChunk(chunk_data);
-      ENVOY_BUG(chunk != nullptr, "Bad partial body callback state");
-      if (common_response.has_header_mutation()) {
-        if (headers_ != nullptr) {
-          const auto mut_status = processHeaderMutation(common_response);
-          if (!mut_status.ok()) {
-            return mut_status;
-          }
-        } else {
-          ENVOY_LOG(debug, "Response had header mutations but headers aren't available");
-        }
-      }
-      if (common_response.has_body_mutation()) {
-        MutationUtils::applyBodyMutations(common_response.body_mutation(), chunk_data);
-      }
-      if (chunk_data.length() > 0) {
-        ENVOY_LOG(trace, "Injecting {} bytes of processed data to filter stream",
-                  chunk_data.length());
-        injectDataToFilterChain(chunk_data, chunk->end_stream);
-      }
-      should_continue = true;
-      clearWatermark();
-      onFinishProcessorCall(Grpc::Status::Ok);
-      partial_body_processed_ = true;
-
-      // If anything else is left on the queue, inject it too
-      if (chunkQueue().receivedData().length() > 0) {
-        const auto& all_data = consolidateStreamedChunks();
-        ENVOY_LOG(trace, "Injecting {} bytes of leftover data to filter stream",
-                  chunkQueue().receivedData().length());
-        injectDataToFilterChain(chunkQueue().receivedData(), all_data.end_stream);
-      }
-    } else {
-      // Fake a grpc error when processor state and received message type doesn't match, beware this
-      // is not an error from grpc.
-      onFinishProcessorCall(Grpc::Status::FailedPrecondition);
-    }
+  auto process_status = processResponseBasedOnState(common_response, should_continue);
```

Comment:
use an explicit type here per style guide.

But i think all this code would be clearer and more concise if you had these functions return `absl::StatusOr<bool>` rather than passing a writeable ref to a bool for should_continue.

Reply:
Agreed